apina-core/src/main/kotlin/fi/evident/apina/utils/PropertyUtils.kt

```diff
@@ -1,8 +1,19 @@
 package fi.evident.apina.utils
 
+import java.util.Locale
+
 fun propertyNameForGetter(getterName: String): String =
     when {
-        getterName.startsWith("get") -> getterName.removePrefix("get").decapitalize()
-        getterName.startsWith("is") -> getterName.removePrefix("is").decapitalize()
+        // Value/inline classes result in name suffixed with hash
+        getterName.contains("-") -> propertyNameForSimpleGetter(getterName.substringBefore('-'))
```

Comment:
Not that it really matters, but as a matter of performance, it's better to search for `char` than a substring. So `getterName.contains('-')`. And then as a matter of style, I'd write that as `'-' in getterName`.

And I do find it somewhat unsatisfying that the code here searching for the delimiter twice, but the code is clear and it _really_ doesn't matter so let's leave it at that. I just couldn't bring myself to write the code this way. :smile:

Reply:
Well now that I think about it, I actually think we should just drop the contains check, because the `substringBefore` does the same, and returns the whole string if it didn't contain it. 😄